dbusmock/templates/accounts_service.py

```diff
@@ -0,0 +1,392 @@
+'''Accounts Service D-Bus mock template'''
+
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the Free
+# Software Foundation; either version 3 of the License, or (at your option) any
+# later version.  See http://www.gnu.org/copyleft/lgpl.html for the full text
+# of the license.
+
+__author__ = 'Marco Trevisan'
+__email__ = 'marco.trevisan@canonical.com'
+__copyright__ = '(c) 2021 Canonical Ltd.'
+__license__ = 'LGPL 3+'
+
+import dbus
+
+from dbusmock import MOCK_IFACE, mockobject
+
+BUS_NAME = 'org.freedesktop.Accounts'
+MAIN_OBJ = '/org/freedesktop/Accounts'
+MAIN_IFACE = 'org.freedesktop.Accounts'
+USER_IFACE = MAIN_IFACE + '.User'
+SYSTEM_BUS = True
+
+
+def get_user_path(uid):
+    return '/org/freedesktop/Accounts/User{}'.format(uid)
+
+
+def load(mock, parameters=None):
+    parameters = parameters if parameters else {}
+    mock.mock_users = {}
+    mock.cached_users = []
+    mock.automatic_login_users = []
+    mock.users_auto_uids = 2000
+
+    mock.AddProperties(MAIN_IFACE, mock.GetAll(MAIN_IFACE))
+
+    for uid, name in parameters.get('users', {}).items():
+        mock.AddUser(uid, name)
+
+
+def emit_properties_changed(mock, interface=MAIN_IFACE, properties=None):
+    if properties is None:
+        properties = mock.GetAll(interface)
+    elif isinstance(properties, str):
+        properties = [properties]
+
+    if isinstance(properties, (list, set)):
+        properties = {p: mock.Get(interface, p) for p in properties}
+    elif isinstance(properties, dict):
+        pass
+
+    mock.EmitSignal(dbus.PROPERTIES_IFACE, 'PropertiesChanged', 'sa{sv}as', (
+        interface, properties, []))
+
+
+@dbus.service.method(MOCK_IFACE, in_signature='xsa{sv}', out_signature='o')
+def AddUser(self, uid, username, overrides=None):
+    '''Add user via uid and username and optionally overriding properties
+
+    Returns the new object path.
+   '''
+    path = get_user_path(uid)
+    default_props = {
+        'Uid': dbus.UInt64(uid),
+        'UserName': username,
+        'RealName': username[0].upper() + username[1:] + ' Fake',
+        'AccountType': dbus.Int32(1),
+        'AutomaticLogin': False,
+        'BackgroundFile': '',
+        'Email': '{}@python-dbusmock.org'.format(username),
+        'FormatsLocale': 'C',
+        'HomeDirectory': '/tmp/dbusmock-home/{}'.format(username),
```

Comment:
The path could also do with a DEFAULT_ constant. Why this odd path? Is it reasonable that test suites would actually create/write into this? Then it shouldn't be a predictable dir in /tmp/, as that's always prone to symlink attacks, existing files, etc. Make it a proper `mkdtemp()`, or something like `/mock/home/` if this is never actually written.

Reply:
All this mock started as a simple tool for testing fprintd locally so didn't bother much in having real scenarios, ideally I just wanted to be to a location that never existed, not sure we should really create a real one at this level though, maybe we can just unset it and leave it to be handled by those who are using the template by overriding if needed?

Reply:
Ack; if it's never written to, please clarify that with /mock/home or /nonexisting/home or something such.